src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -48,32 +46,121 @@ final case class EServicesApiServiceImpl(
     CatalogProcess.EServiceDescriptorState.DEPRECATED
   )
 
-  override def getEserviceDescriptor(eserviceId: String, descriptorId: String)(implicit
+  private val SUBSCRIBED_AGREEMENT_STATES: Set[AgreementProcess.AgreementState] = Set(
+    AgreementProcess.AgreementState.PENDING,
+    AgreementProcess.AgreementState.ACTIVE,
+    AgreementProcess.AgreementState.SUSPENDED
+  )
+
+  override def getCatalogEServiceDescriptor(eserviceId: String, descriptorId: String)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerEServiceDescriptor: ToEntityMarshaller[EServiceDescriptor],
-    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCatalogEServiceDescriptor: ToEntityMarshaller[CatalogEServiceDescriptor]
   ): Route = {
-    val result: Future[EServiceDescriptor] = for {
+    val result: Future[CatalogEServiceDescriptor] = for {
+      requesterId                    <- getOrganizationIdFuture(contexts)
       (eserviceUUID, descriptorUUID) <- eserviceId.toFutureUUID.zip(descriptorId.toFutureUUID)
       eService                       <- catalogProcessService.getEServiceById(eserviceUUID)
       descriptor                     <- eService.descriptors
         .find(_.id === descriptorUUID)
-        .toFuture(EServiceDescriptorNotFound(eserviceId, descriptorId))
-      mail                           <- tenantManagementService
-        .getTenant(eService.producer.id)
-        .map(_.mails.headOption.map(_.toApi))
-    } yield descriptor.toApi(mail)
+        .toFuture(EServiceDescriptorNotFound(eService.id.toString, descriptorId))
+      attributes <- attributeRegistryManagementService.getBulkAttributes(extractIdsFromAttributes(eService.attributes))(
+        contexts
+      )
+      requesterTenant <- tenantManagementService
+        .getTenant(eService.producerId)
+      agreement       <- agreementProcessService
+        .getAgreements(
+          consumerId = requesterId.some,
+          eServiceId = eserviceId.some,
+          descriptorId = descriptorId.some,
+          states = Seq.empty,
+          latest = None
+        )
+        .map(_.headOption)
+    } yield CatalogEServiceDescriptor(
```

Comment:
I don't like this verbosity here, but I'm not sure that moving this piece of code would be the solution for more readability, the problem is the number of fields retrieved.
Let me know your opinion.

Reply:
I can't tell which way we can follow to reduce the verbosity if not masking the call behind methods calls that implement a specific way of searching, i.e.
`getDescriptor(requesterId: String, eserviceId: String, descriptorId: String) = ???`
but that's not a solution.

Maybe doing this 👇 will at least reduce the line number :trollface: 
```diff
- tenantManagementService
-         .getTenant(eService.producerId)
+ tenantManagementService.getTenant(eService.producerId)
```

```diff
- agreementProcessService
-        .getAgreements(
+ agreementProcessService.getAgreements(
```

Reply:
🤣